script/testing/util/test_server.py

```diff
@@ -153,6 +159,20 @@ def run_test(self, test_case: TestCase):
         # run the pre test tasks
         test_case.run_pre_test()
 
+        # start a thread to collect the memory info if needed
+        if self.collect_mem_info:
+            # spawn a thread to collect memory info
+            self.collect_mem_thread = threading.Timer(
+                self.collect_mem_freq, update_mem_info, [
+                    self.db_process.pid, self.collect_mem_freq,
+                    test_case.mem_info_dict
+                ])
+            # collect the initial memory info
+            update_mem_info(self.db_process.pid, self.collect_mem_freq,
+                            test_case.mem_info_dict)
+            # collect the subsequent memory info
+            self.collect_mem_thread.start()
```

Comment:
I'm pretty sure this isn't going to do what you want. We need to break up the oltpbench test execution commands. There is a data loading stage in OLTPBench and we don't want to gather the memory info during that, we just want to gather the memory info during the test. I think the flow would be 1. oltpbench load data 2. create mem_info thread 3. start OLTPBench test 4. start mem_info thread 5. When OLTPBench test complete make sure the mem_info is also complete

Reply:
Checkout the `--load` and `--execute` commands https://github.com/oltpbenchmark/oltpbench/wiki I could be wrong but I think we are doing all the commands at once. So I think this might require some more significant changes to the OLTPBenchTestCase code.

Reply:
Done. Moved the create and load DB actions into pretest stage.